app/models/user.rb

```diff
@@ -76,6 +75,11 @@ def to_s
     twitch_display_name || name || 'somebody'
   end
 
+  def rivals
+    User.joins('INNER JOIN rivalries ON rivalries.to_user_id = users.id AND rivalries.from_user_id = ' + id.to_s)
+        .select('users.*, rivalries.category_id')
```

Comment:
What's the purpose of this `select`?

Reply:
To have access to `category_id` in the returned relation object.  Since rivalries aren't global like twitch follows, they only make sense within the context of a category.  Otherwise we would get back a list of users and have no way to tell which user goes to which run.  This way we get back a list of all users (which might have more than one of the same user), and then be able to iterate on them and figure out what category to then look up easily.

Reply:
Ok I'm confused. I understand the need to correlate a User with a Category, but doesn't this `User#rivals` function return an array of ActiveRecord `User`s? But [later][1] I see you calling both `.category_id` (a `Rivalry` method) and `.pb_for` (a `User` method) on them. What type of object are these?

[1]: https://github.com/glacials/splits-io/pull/420/files#diff-b071ff096d8d990198c6eb43c1321faeR46

Reply:
It returns an array of `User`s that have the `category_id` that they are a rival in embeded in them.  That is why we do the weird select statement `.select('users.*, rivalries.category_id')`.  It selects all fields from the user, and the `category_id` field from the rivalries table and merges it into one record.

Reply:
Whoa. I had no idea that was a thing Rails could do. It's like a... frankenstein object. Not sure how I feel about that, lol. Is it possible / still efficient to pass around `Rivalry`s as an alternative?

Reply:
With this pr, it never loads any of the `Rivalry` objects into memory, it just loads your rival user frankenstein objects.  It could probably have `user.rivals` return a tuple of the `User` and `Rival` objects, with only one more db request (one to actually retrieve all rivals), and then zip them with the users (just remove the select statement from current code).  I'm not sure how clean that would be though.  Up to you on what you think is best.

Reply:
Yeah, nevermind I'm just retarded.  Adds one sql query but is much cleaner.

Reply:
lol ok nice